Add support for Anthropic models

zakkor opened the pull request:
(wip -- tool use support has not been added yet)

zakkor committed:
Add support for Anthropic models (wip -- tool use support has not been added yet)

LinqLover commented on packages/SemanticText.package/AnthropicConversationModel.class/instance/maxCompletionTokens.st:
```diff
@@ -0,0 +1,11 @@
+	"https://docs.anthropic.com/en/docs/about-claude/models"
+	"Claude 3.5 family all have 8192 max tokens output, Claude 3 family have 4096."
+	
+	(self resolvedName beginsWith: 'claude-3-5-')
+		ifTrue: [^ 8192]
+		ifFalse: [^ 4096].
```
very small detail:
```suggestion
	^ (self resolvedName beginsWith: 'claude-3-5-')
		ifTrue: [8192]
		ifFalse: [4096]
```

:+1:/:-1: whether you would like to receive feedback like that or not :)

zakkor replied:
Yep!

LinqLover commented on packages/SemanticText.package/AnthropicConversationModel.class/instance/resolvedName.st:
```diff
@@ -0,0 +1,6 @@
+resolvedName
+
+	self flag: #modelConstants.
+	"Anthropic define '-latest' models directly, for which we already have definitions. No need to resolve anything."
```
Yeah, OpenAI defines these directly, too. It was meant to duplicate `centsForCompletionToken` et al. so you don't have to repeat the numbers for the latest pinned model twice. Should probably explain this in a comment ^^

zakkor replied:
Ah, I understand now. Will fix that!

zakkor committed:
Add `asAnthropicObject` extensions in Semantic* classes, and use this in the Anthropic classes instead of asOpenAIObject;
Remove rate limit data from Anthropic models and add a note that they're unimplemented;
Make `getAnswers: number` with a number > 1 return an array of completions instead of throwing an error (I am not sure if number=1 should return an array of size 1 or just the completion itself).

zakkor committed:
Update packages/SemanticText.package/AnthropicConversationModel.class/instance/maxCompletionTokens.st

Co-authored-by: Christoph Thiede <38782922+LinqLover@users.noreply.github.com>

zakkor committed:
Make tool calls (non-streamed) work

zakkor committed:
Add `topKSampling` (Anthropic only). Cleanup, rename things to match more closely what the Anthropic API sends us, handle more streaming events in preparation for streaming tool calls.
Fix rawOutput tracking for Anthropic models. Fix an issue with getting multiple assistant replies.

zakkor committed:
Add support for system messages

zakkor committed:
Add streaming tool call implementation for Anthropic models
